tests/test_derivatives.py

```diff
@@ -731,6 +732,12 @@ def test_issue_2442(self):
         dfdxdy_split = f.dxc.dyc
         assert dfdxdy.evaluate == dfdxdy_split.evaluate
 
+    def test_cross_newnest(self):
+        grid = Grid((11, 11))
+        f = Function(name="f", grid=grid, space_order=2)
+
+        assert f.dxdy == f.dx.dy
```

Comment:
Would be good to test that chaining derivatives with various `x0` leads to sensible consolidation too.

Reply:
https://github.com/devitocodes/devito/pull/2458/files#diff-f3c30bdc08b83267e30f815ee9e9a8e1ff22ad2b694efc6813652af0f2557869R395